mapie/utils.py

```diff
@@ -654,3 +666,369 @@ def compute_quantiles(vector: NDArray, alpha: NDArray) -> NDArray:
                 ]
             )[:, 0]
     return quantiles_
+
+
+def check_calib_set(
+    X: ArrayLike,
+    y: ArrayLike,
+    sample_weight: Optional[NDArray] = None,
+    X_calib: Optional[ArrayLike] = None,
+    y_calib: Optional[ArrayLike] = None,
+    sample_weight_calib: Optional[NDArray] = None,
+    calib_size: Optional[float] = 0.3,
+    random_state: Optional[Union[int, np.random.RandomState]] = None,
+    shuffle: Optional[bool] = True,
+    stratify: Optional[ArrayLike] = None,
+) -> Tuple[
+    ArrayLike, ArrayLike, ArrayLike, ArrayLike,
+    Optional[NDArray], Optional[NDArray]
+]:
+    """
+    Check if a calibration set has already been defined, if not, then
+    we define one using the `train_test_split` method.
+
+    Parameters
+    ----------
+    Same definition of parameters as for the ``fit`` method.
+
+    Returns
+    -------
+    Tuple[
+        ArrayLike, ArrayLike, ArrayLike, ArrayLike,
+        Optional[NDArray], Optional[NDArray]
+    ]
+    - [0]: ArrayLike of shape (n_samples_*(1-calib_size), n_features)
+        X_train
+    - [1]: ArrayLike of shape (n_samples_*(1-calib_size),)
+        y_train
+    - [2]: ArrayLike of shape (n_samples_*calib_size, n_features)
+        X_calib
+    - [3]: ArrayLike of shape (n_samples_*calib_size,)
+        y_calib
+    - [4]: Optional[NDArray] of shape (n_samples_*(1-calib_size),)
+        sample_weight_train
+    - [5]: Optional[NDArray] of shape (n_samples_*calib_size,)
+        sample_weight_calib
+    """
+    sw_calib = sample_weight_calib
+    if X_calib is None or y_calib is None:
+        if sw_calib is not None:
+            warnings.warn(
+                "WARNING: sample weight for calibration"
+                + " were provided without X_calib and y_calib."
+            )
+        if sample_weight is None:
+            (
+                X_train, X_calib, y_train, y_calib
+            ) = train_test_split(
+                    X,
+                    y,
+                    test_size=calib_size,
+                    random_state=random_state,
+                    shuffle=shuffle,
+                    stratify=stratify
+            )
+            sw_train = sample_weight
+        else:
+            (
+                    X_train,
+                    X_calib,
+                    y_train,
+                    y_calib,
+                    sw_train,
+                    sw_calib,
+            ) = train_test_split(
+                    X,
+                    y,
+                    sample_weight,
+                    test_size=calib_size,
+                    random_state=random_state,
+                    shuffle=shuffle,
+                    stratify=stratify
+            )
+    else:
+        X_train, y_train, sw_train = X, y, sample_weight
+    X_train, X_calib = cast(ArrayLike, X_train), cast(ArrayLike, X_calib)
+    y_train, y_calib = cast(ArrayLike, y_train), cast(ArrayLike, y_calib)
+    return X_train, y_train, X_calib, y_calib, sw_train, sw_calib
```

Comment:
As mentioned in calibration.py, I think this function could be way simpler if the user only give X and Y

Reply:
done.